crates/runtime/src/lib.rs

```diff
@@ -91,8 +90,18 @@ pub unsafe trait Store {
         &mut self,
     ) -> (&mut VMExternRefActivationsTable, &dyn ModuleInfoLookup);
 
-    /// Returns a reference to the store's limiter for limiting resources, if any.
-    fn limiter(&mut self) -> Option<&mut dyn ResourceLimiter>;
+    /// Callback invoked to allow the store's resource limiter to reject a memory grow operation.
+    fn limiter_memory_growing(
+        &mut self,
+        current: usize,
+        desired: usize,
+        maximum: Option<usize>,
+    ) -> bool;
+    /// Callback invoked to notify the store's resource limiter that a memory grow operation has
+    /// failed.
+    fn limiter_memory_grow_failed(&mut self, error: &anyhow::Error);
+    /// Callback invoked to allow the store's resource limiter to reject a table grow operation.
+    fn limiter_table_growing(&mut self, current: u32, desired: u32, maximum: Option<u32>) -> bool;
```

Comment:
I think it's probably ok to skip the `limiter_` prefix on these methods (micro-nit)